scripts/fix/dry-buckets.lua

```diff
@@ -0,0 +1,22 @@
+-- Removes water from buckets (for lye-making).
+--[[=begin
+
+fix/dry-buckets
+===============
+Removes water from all buckets in your fortress, allowing them
+to be used for making lye.  Skips buckets in buildings (eg a well),
+being carried, or currently used by a job.
+
+=end]]
+
+local emptied = 0
+local water_type = dfhack.matinfo.find('WATER').type
+
+for _,item in ipairs(df.global.world.items.all) do
+    if item:getMaterial() == water_type and not (item.flags.in_job or item.flags.in_building) then
```

Comment:
item:getType() is not a material type "get". It's fundamental item class type "get" so it should be okay for all the items.